indigo/analysis/toc/base.py

```diff
@@ -384,6 +384,23 @@ def as_dict(self):
         }
 
 
+class BeautifulElement:
```

Comment:
We can make each instance of this class act as a proxy for the TOCElement object it wraps. That way, callers don't have to know that some information comes from this object, and some from the proxy.

The idea is to use `__getattr__` to intercept calls to missing attributes, and pass them on to the proxied object. See https://blog.rmotr.com/python-magic-methods-and-getattr-75cf896b3f88

```
class BeautifulElement:
  ... (init as before)...

  def __getattr__(self, attr):
    return getattr(self.toc_element, attr)
```

Then you can do both `beautiful.num` (defined on the beautiful element) and `beautiful.id` (defined on the wrapped `toc_element`).

Reply:
incredible! thank you